apps/condo/domains/meter/components/CreateMeterReadingsActionBar.tsx

```diff
@@ -41,11 +48,17 @@ export const CreateMeterReadingsActionBar = ({
         >
             {
                 ({ getFieldsValue, getFieldValue, getFieldError }) => {
-                    let isSubmitButtonDisabled, isCreateMeterButtonDisabled, errors, requiredErrorMessage
+                    let isSubmitButtonDisabled, errors, requiredErrorMessage
+
+                    const metersByIds = meters.reduce((result, meter) => ({ ...result, [meter.id]: meter }), {})
```

Comment:
Use `Object.assign` or default `for` loop.

`...result` will copy 1, then 2, then 3, then 4 objects and so on. So its `N^2` copied objects

Reply:
Here will be tiny arrays. But ok, replaced by for-loop.